### PR TITLE
[module] add functions for ElixirModule

### PR DESCRIPTION
**Pull request by 1SAA**
* add __init__ functions, inluding __init_group, __init_fetcher
* add state_dict function for unit test
* add a unit test for __init__ of ElixirModule

### CODE REVIEW

File: elixir/search/result.py [hunk 1]
```diff
@@ -1,15 +1,24 @@
-from typing import Dict, List
+from typing import Dict, List, NamedTuple
+
+import torch
 
 from elixir.chunk import ChunkGroup
 
 
+class ChunkPlan(NamedTuple):
+    name_list: List[str]
```
Comment (FrankLeeeee) [inline elixir/search/result.py:9]: What is a ChunkPlan? Add some docstring for clarity.

**Comment by FrankLeeeee** [comment]
Can you paste a screenshot of your test results?

File: elixir/wrapper/module.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+def get_param_optim_data(param_data: torch.Tensor, param_dtype: torch.dtype):
+    param_data = param_data.to(gpu_dev())
+    optim_data = param_data.clone() if param_data.dtype == torch.float else param_data.float()
+    param_data = param_data.to(param_dtype)
+    return param_data, optim_data
+
+
+class ElixirModule(nn.Module):
+
```
Comment (FrankLeeeee) [inline elixir/wrapper/module.py:27]: Do take of documentation for maintainability. Add docstrings whenever appropriate.

File: elixir/wrapper/module.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    data = torch.randn(2, 1024, dtype=torch.float16)
+    y = x.state_dict(keep_vars=True)
+    for name, tensor in y.items():
+        if isinstance(tensor, nn.Parameter):
+            print('parameter', name)
+            tensor.data = tensor.data.to(torch.float16)
+        else:
+            print('buffer', name)
+    print(y)
+    print(x(data))
```
Comment (FrankLeeeee) [inline elixir/wrapper/module.py:209]: Don't put this test here.

File: elixir/search/result.py (hunk 1 above)
Comment (1SAA) [reply elixir/search/result.py]: Done

**Comment by 1SAA** [comment]
image

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**